src/mlpack/methods/ann/activation_functions/softplus_function.hpp

```diff
@@ -51,8 +51,10 @@ class SoftplusFunction
    */
   static double Fn(const double x)
   {
+    const double val = std::log(1 + std::exp(x));
     if (x < DBL_MAX)
-      return x > -DBL_MAX ? std::log(1 + std::exp(x)) : 0;
+      return x > -DBL_MAX ? (std::isnan(val) || std::isinf(val) ? x : val)
+          : 0;
```

Comment:
I'd much prefer simple `if-else` loops in place of nested conditionals. Improves readability.